src/sb_laserscan_processing/src/ConeIdentification.cpp

```diff
@@ -0,0 +1,293 @@
+/**
+ * Created by William Gu on 24/03/18
+ * Implementation of Cone Identification methods for identifying cones from a laser msg
+ */
+
+#include <ConeIdentification.h>
+
+
+std::vector<mapping_igvc::ConeObstacle> ConeIdentification::identifyCones(const sensor_msgs::LaserScan &laser_msg, double dist_tol, double radius_exp, double radius_tol, int line_point_dist, double ang_threshold){
+    std::vector<mapping_igvc::ConeObstacle> identified_cones;
+    std::vector<mapping_igvc::Point2D> edge_points; //Represents points in a potential cluster
+
+    int numIndices = (laser_msg.angle_max - laser_msg.angle_min) / laser_msg.angle_increment;
+    for (int i=0; i<numIndices; i++){
+        if (laser_msg.ranges[i] > laser_msg.range_max || laser_msg.ranges[i] < laser_msg.range_min) { //Check if curr laserscan point is in invalid range
+
+            if (edge_points.size() >= line_point_dist * 2 + 1) { // Don't bother splitting edges with too few points
+                std::vector<std::vector<mapping_igvc::Point2D>> split_edges = splitEdge(edge_points, line_point_dist, ang_threshold); //Split edge points if multiple cones detected (Temp values right now as params)
+                for (int i = 0; i < split_edges.size(); i++){
+                    
+                    mapping_igvc::ConeObstacle potential_cone = edgeToCone(split_edges[i]);
+
+                    //Testing
+                    /*
+                    for (int j=0 ;j < split_edges[i].size(); j++){
+                        std::cout<<"("<<split_edges[i][j].x<<","<<split_edges[i][j].y<<")"<<std::endl; //Print points in edge
+                    }*/
```

Comment:
In general, we should try and remove commented out code and/or testing code (though I understand this is a WIP).